src/wallet/asyncrpcoperation_shieldcoinbase.cpp

```diff
@@ -283,7 +284,34 @@ bool ShieldToAddress::operator()(const libzcash::UnifiedAddress &uaddr) const {
     // TODO check if an Orchard address is present, send to it if so.
     const auto receiver{uaddr.GetSaplingReceiver()};
     if (receiver.has_value()) {
-        return ShieldToAddress(m_op, sendAmount)(receiver.value());
```

Comment:
I think that what we should do here is actually pass the `std::optional<UnifiedAddress>` to the `ShieldToAddress` constructor; that will avoid the code duplication for now. Alternately, factor out the `Sapling` case as a static method on the `ShieldToAddress` class and use that in both circumstances. Also, we should at very least have a ticket to fix the OVK issue.